src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -295,7 +301,62 @@ public static Complex Exp(Complex value)
         [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly", MessageId = "Sqrt", Justification = "Sqrt is the name of a mathematical function.")]
         public static Complex Sqrt(Complex value)
         {
-            return FromPolarCoordinates(Math.Sqrt(value.Magnitude), value.Phase / 2.0);
+
+            if (value._imaginary == 0.0)
+            {
+                // Handle the trivial case quickly.
+                if (value._real < 0.0)
+                {
+                    return new Complex(0.0, Math.Sqrt(-value._real));
+                }
+                else
+                {
+                    return new Complex(Math.Sqrt(value._real), 0.0);
+                }
+            }
+            else
```

Comment:
Agreed. (Had to laugh, though, since over the course of my career I have got about equal numbers of comments saying "early return makes this else superfluous" and "don't rely on early returns; always make else explicit".) Will change if another PR update is required, but not otherwise.